docs: Update supported models (#389)
foundation-model-stack/fms-hf-tuning: README.md

## Supported Models

- For each tuning technique, we run testing on a single large model of each architecture type and claim support for the smaller models. For example, with QLoRA technique, we tested on granite-34b GPTBigCode and claim support for granite-20b-multilingual.

- LoRA Layers supported : All the linear layers of a model + output `lm_head` layer. Users can specify layers as a list or use `all-linear` as a shortcut. Layers are specific to a model architecture and can be specified as noted [here](https://github.com/foundation-model-stack/fms-hf-tuning?tab=readme-ov-file#lora-tuning-example)

- Legend:

  ✅ Ready and available 

  ✔️ Ready and available - compatible architecture (*see first bullet point above)

  🚫 Not supported

  ? May be supported, but not tested

Model Name & Size  | Model Architecture | Full Finetuning | Low Rank Adaptation (i.e. LoRA) | qLoRA(quantized LoRA) | 
-------------------- | ---------------- | --------------- | ------------------------------- | --------------------- |
Granite 3.0 3B       | GraniteForCausalLM     | ✅* | ✅* | ✅* |

> Granite 3.0 8B       | GraniteForCausalLM     | ✅* | ✅* | ✔️  |

Reviewer: Since we are not mentioning the power models then the correct family of granite3 models are:
> IBM's Granite 3.0 family includes:
> - General Purpose/Language: Granite 3.0 8B Instruct, Granite 3.0 2B Instruct, Granite 3.0 8B Base, Granite 3.0 2B Base
> - Guardrails & Safety: Granite Guardian 3.0 8B, Granite Guardian 3.0 2B
> - Mixture-of-Experts: Granite 3.0 3B-A800M Instruct, Granite 3.0 1B-A400M Instruct, Granite 3.0 3B-A800M Base, Granite 3.0 1B-A400M Base

So to match this list 
Granite 3.0 2B       | GraniteForCausalLM     | ✔️* | ✔️* | ✔️* |
Granite 3.0 8B       | GraniteForCausalLM     | ✅* | ✅* | ✔️  |

unless we are mentioning the power models in which:
Granite PowerLM 3B       | GraniteForCausalLM     | ✅* | ✅* | ✅* |
Granite 3.0 8B       | GraniteForCausalLM     | ✅* | ✅* | ✔️  |

## Supported Models

> Current supported and tested models are `Granite v3`, `Llama3` (8B configuration has been tested) and `GPTBigCode`.

Reviewer 2: ```suggestion
Current supported and tested models are `Granite 3.0`, `Llama3` (8B configuration has been tested) and `GPTBigCode`.
```

Reviewer: The above is also not fully accurate as it leaves out the mistral, mixtral, and granite with llama models (small code models) which wouldn't be distinguished with llama3 or granite3 or gptbigcode.

- For each tuning technique, we run testing on a single large model of each architecture type and claim support for the smaller models. For example, with QLoRA technique, we tested on granite-34b GPTBigCode and claim support for granite-20b-multilingual.

- LoRA Layers supported : All the linear layers of a model + output `lm_head` layer. Users can specify layers as a list or use `all-linear` as a shortcut. Layers are specific to a model architecture and can be specified as noted [here](https://github.com/foundation-model-stack/fms-hf-tuning?tab=readme-ov-file#lora-tuning-example)

> - Legend:
> 
> ✅ Ready and available 
> 
> ✔️ Ready and available - compatible architecture (*see first bullet point above)
> 
> 🚫 Not supported
> 
> ?  may be supported, but not tested

Reviewer 2: ```suggestion
- Legend:

   ✅ Ready and available 

   ✔️ Ready and available - compatible architecture (*see first bullet point above)

   🚫 Not supported

   ?  may be supported, but not tested
```
Could indent these out a bit so they are inline with the text "Legend"

Reviewer: I'm wondering if we need the ? since it's not being used in the table

Model Name & Size  | Model Architecture | Full Finetuning | Low Rank Adaptation (i.e. LoRA) | qLoRA(quantized LoRA) | 
-------------------- | ---------------- | --------------- | ------------------------------- | --------------------- |

> Granite 3B           | Granite v3       | ✅*             | ✅*                              | ✅*                   | 
> Granite 8B           | Granite v3       | ✅*             | ✅*                              | ✔️                     | 

Reviewer 2: ```suggestion
Granite 3.0 3B           | GraniteForCausalLM       | ✅*             | ✅*                              | ✅*                   | 
Granite 3.0 8B           | GraniteforCausalLM       | ✅*             | ✅*                              | ✔️                     | 
```

Reviewer: ^^agreed with above change

[Granite 3B](https://huggingface.co/ibm-granite/granite-3b-code-base) | LlamawithCausalLM          | ✅ | ✔️  | ✔️ | 
[Granite 8B](https://huggingface.co/ibm-granite/granite-8b-code-base) | LlamawithCausalLM          | ✅ | ✅ | ✅ |
Granite 13B                                                           | GPTBigCodeForCausalLM      | ✅ | ✅ | ✔️  | 
[Granite 20B](Granite-20b-multilingual-base)                          | GPTBigCodeForCausalLM      | ✅ | ✔️  | ✔️  | 

> [Granite 34B](https://huggingface.co/ibm-granite/granite-34b-code-instruct/tree/main) | GPTBigCodeForCausalLM | 🚫 | ✅ | ✅ | 

Reviewer: Are the models we had tested on the 20b multilingual and 34b instruct? I see there is also the base model for 34b https://huggingface.co/ibm-granite/granite-34b-code-base-8k and 20b there is also a base model as well as code models. We may not need to be specific about which one we test?

Llama3.1-8B                         | LLaMA 3.1          | ✅ - supported from platform up to 8k context length - same architecture as llama3-8b  | ✔️ | ✔️ |  
Llama3.1-70B(same architecture as llama3) | LLaMA 3.1 | 🚫 - same as Llama3-70B | ✔️  | ✔️ | 

> Llama3.1-405B                             | LLaMA 3.1 | 🚫 | 🚫 | 🚫 - issue loading model on 8 GPUs | 

Reviewer: We did get a successful run for qlora for 405b so can change to ✅

Author: so `Llama3.1-405B ` only has qLora supported?, FT and Lora are `🚫`?

Reviewer: correct since the model is so large it only worked with qlora

Llama3-8B                                 | LLaMA 3   | ✅ | ✅ | ✔️ |  
Llama3-70B                                | LLaMA 3   | 🚫 | ✅ - had some issues loading the merged model, haven’t retested recently | ✅ | 
Mixtral 8x7B                              | Mixtral   | ✅ | ✅ | ✅ |
Mistral-7b                                | Mistral   | ✅ | ✅ | ✅ |  
Mistral large                             | Mistral   | 🚫 | 🚫 | 🚫 | 

> aLLaM-13b                                 | LlamaForCausalLM |  ✅ | ✅ | ✅ |

Reviewer: Please move the aLLaM model up near llama since its the same model architecture, but was trained on arabic

## Supported Models

- For each tuning technique, we run testing on a single large model of each architecture type and claim support for the smaller models. For example, with QLoRA technique, we tested on granite-34b GPTBigCode and claim support for granite-20b-multilingual.

- LoRA Layers supported : All the linear layers of a model + output `lm_head` layer. Users can specify layers as a list or use `all-linear` as a shortcut. Layers are specific to a model architecture and can be specified as noted [here](https://github.com/foundation-model-stack/fms-hf-tuning?tab=readme-ov-file#lora-tuning-example)

- Legend:

  ✅ Ready and available 

  ✔️ Ready and available - compatible architecture (*see first bullet point above)

  🚫 Not supported

  ? May be supported, but not tested

Model Name & Size  | Model Architecture | Full Finetuning | Low Rank Adaptation (i.e. LoRA) | qLoRA(quantized LoRA) | 
-------------------- | ---------------- | --------------- | ------------------------------- | --------------------- |
Granite PowerLM 3B   | GraniteForCausalLM | ✅* | ✅* | ✅* |
Granite 3.0 2B       | GraniteForCausalLM | ✔️* | ✔️* | ✔️* |

> Granite 3.0 8B       | GraniteForCausalLM | ✅* | ✅* | ✔️ |

Reviewer: Looks like granite 3.0 8B is in here twice?

Reviewer: otherwise the rest of the changes look good to me
